spec/models/rdv_spec.rb

```diff
@@ -65,26 +65,88 @@
 
   describe "#cancellable_by_user?" do
     let(:now) { Time.zone.parse("2021-05-03 14h00") }
+    let(:rdv) { build :rdv, starts_at: starts_at, motif: motif }
+    let(:starts_at) { now + 5.hours }
+    let(:motif) { build(:motif, rdvs_cancellable_by_user: true) }
 
     before { travel_to(now) }
 
-    context "when Rdv starts in 5 hours" do
-      let(:rdv) { create(:rdv, starts_at: now + 5.hours) }
-
+    context "when Rdv starts in more than 4 hours" do
       it { expect(rdv.cancellable_by_user?).to eq(true) }
+    end
 
-      context "but is already cancelled" do
-        let(:rdv) { create(:rdv, status: "excused", starts_at: now + 5.hours) }
+    context "when Rdv starts in 4 hours or less" do
+      let(:starts_at) { now + 4.hours }
 
-        it { expect(rdv.cancellable_by_user?).to eq(false) }
-      end
+      it { expect(rdv.cancellable_by_user?).to eq(false) }
+    end
+
+    context "when it is already cancelled" do
+      let(:rdv) { build(:rdv, status: "excused", starts_at: starts_at, motif: motif) }
+
+      it { expect(rdv.cancellable_by_user?).to eq(false) }
     end
 
-    context "when Rdv starts in 4 hours" do
-      let(:rdv) { create(:rdv, starts_at: now + 4.hours) }
+    context "when it is collectif" do
+      let(:motif) { build(:motif, collectif: true) }
 
       it { expect(rdv.cancellable_by_user?).to eq(false) }
     end
+
+    context "when the rdvs are set as not cancellable by the user in the motif" do
+      let(:motif) { build(:motif, rdvs_cancellable_by_user: false) }
+
+      it { expect(rdv.cancellable_by_user?).to eq(false) }
+    end
+  end
+
+  describe "#editable_by_user?" do
+    let(:now) { Time.zone.parse("2021-05-03 14h00") }
+    let(:rdv) { build :rdv, starts_at: starts_at, motif: motif, created_by: "user" }
+    let(:starts_at) { now + 3.days }
+    let(:motif) { build(:motif, rdvs_editable_by_user: true, reservable_online: true) }
+
+    before { travel_to(now) }
+
+    context "when Rdv starts in more than 2 days" do
+      it { expect(rdv.editable_by_user?).to eq(true) }
+    end
```

Comment:
Une raison particulière de ne pas utiliser la forrme "`subject` + `is_expected`" ? 

```suggestion
    subject(:editable_by_user?) { rdv.editable_by_user? }

    context "when Rdv starts in more than 2 days" do
      it { is_expected.to eq(true) }
    end
```

Je suppose que c'est une question de goût, pour éviter l'indirection. J'aime bien ton approche en fait, je suis curieux d'avoir ton avis. :smile_cat:

Reply:
Alors non y a pas de raison particulière, il est vrai que j'utilise rarement le helper `is_expected`, mais c'est vraiment une question d'habitude :). 
Et oui pour moi c'est un peu plus clair comme ça mais j'ai clairement pas d'avis tranché là-dessus (donc je peux changer au pire des cas si vous voulez avoir quelque chose d'homogène partout) !

Reply:
On a pas de préférence au sein de l'équipe je pense, et pour ma part j'aime bien ta version (qui dévie de celle des puristes de rspec, mais on est plus en 2009) ! :wink:

Reply:
Alors pour mettre mon grain de sel, moi j'aime bien les `it` avec tout ce qu'il faut dedans, et je ne suis pas fan des `let` et autres `before`... 

Ce qu'on peut retenir pour l'instant, c'est que nous n'avons pas de pratique très uniformisé dans nos tests... Ça serait sans doute quelque chose à faire ?